packages/as-types/assembly/amount.ts

```diff
@@ -1,202 +0,0 @@
-    }
-
-    if (other.value > this.value) {
-      // tests underflow
-      return new Result(
-        new Amount(),
-        'the difference is impossible: underflow',
-      );
-    }
-
-    return new Result(new Amount(this.value - other.value, this.currency));
-  }
-
-  /**
-   * Check if existent amount is lower than given one.
-   *
-   * @remarks
-   * Comparison between amounts with different {@link Currency} will result in returning false.
-   *
-   * @param  other - Amount to check against.
-   *
-   * @returns true if the amount is lower than the given one.
-   */
-  @operator('<')
-  lessThan(other: Amount): bool {
-    if (this.currency != other.currency) {
-      return false;
-    }
-    return this.value < other.value;
-  }
-
-  /**
-   * Deserializes an Amount from an {@link Args} 'array of bytes'.
-   *
-   * @see {@link Result} for more information about the return type.
-   *
-   * @param args - Argument to deserialize.
-   *
-   * @returns
-   * A Result object containing:
-   * - The deserialized Amount if succeeded.
-   * - An error message if:
-   *  - if there is an error with deserializing the 'value'.
-   *  - if there is an error with deserializing the 'currency'.
-   */
-  static fromArgs(args: Args): Result<Amount> {
-    const value = args.nextU64();
-    if (value.isErr()) {
-      return new Result(new Amount(), 'deserializing Amount: ' + value.error!);
-    }
-
-    const currency = Currency.fromArgs(args);
-    if (currency.isErr()) {
-      return new Result(
-        new Amount(),
-        'deserializing Currency: ' + currency.error!,
-      );
-    }
-
-    return new Result(new Amount(value.unwrap(), currency.unwrap()));
-  }
-
-  /**
-   * Serializes and adds the Amount to the given serialized {@link Args}.
-   *
-   * @param args -The arguments to add the serialized Amount to.
-   */
-  addArgs(args: Args): void {
-    args.add(this.value);
-    this.currency.addArgs(args);
-  }
-
-  /**
-   * Serializes the Amount to a new {@link Args} object.
-   *
-   * @returns The serialized Amount as {@link Args}.
-   */
-  toArgs(): Args {
-    const args = new Args();
-
-    this.addArgs(args);
-
-    return args;
-  }
-
-  /**
-   * Tests if two amounts are identical.
-   *
-   * @remarks
-   * Two amounts are identical if they have the same value as well as the same currency type!
-   *
-   * @param other - Amount to check against.
-   *
-   * @returns true if the amounts are identical.
-   */
-  @operator('==')
-  equals(other: Amount): boolean {
-    return this.currency == other.currency && this.value == other.value;
-  }
-
-  /**
-   * Tests if two amounts are different.
-   *
-   * @remarks
-   * Two amounts are different if they have the same currency type even if they have the same value!
-   *
-   * @param other - Amount to check against.
-   *
-   * @returns true if the amounts are different.
-   */
-  @operator('!=')
-  notEqual(other: Amount): boolean {
-    return !(this == other);
-  }
-}
```

Comment:
@Elli610 Why did you remove the content of this file? same for Result

Reply:
I didn't. It wasn't deleted before the rebase